src/App.jsx

```diff
@@ -14,33 +15,88 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState('');
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const handleSortAlphabetically = () => {
+    const sortedGoods = [...goodsFromServer].sort();
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    setGoods(isReversed ? sortedGoods.reverse() : sortedGoods);
+    setSortType('alphabet');
+    setIsReversed(false);
+  };
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  const handleSortByLength = () => {
+    const sortedGoods = [...goodsFromServer].sort(
+      (a, b) => a.length - b.length,
+    );
+
+    setGoods(isReversed ? sortedGoods.reverse() : sortedGoods);
+    setSortType('length');
+    setIsReversed(false);
+  };
+
+  const handleReverse = () => {
+    setGoods([...goods].reverse());
+    setIsReversed(!isReversed);
+  };
+
+  const handleReset = () => {
+    setGoods(goodsFromServer);
+    setSortType('');
+    setIsReversed(false);
+  };
+
+  const isResetVisible =
+    JSON.stringify(goods) !== JSON.stringify(goodsFromServer);
```

Comment:
Comparing arrays with JSON.stringify might not be the best approach as it's not the most performant and it can lead to false negatives. A better approach would be to compare the lengths of the arrays and then compare each element individually.